anvil/tests/it/api.rs

```diff
@@ -71,3 +75,25 @@ async fn can_get_network_id() {
     let chain_id = provider.get_net_version().await.unwrap();
     assert_eq!(chain_id, CHAIN_ID.into());
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn can_get_block_by_number() {
+    let (_api, handle) = spawn(NodeConfig::test().port(next_port())).await;
+    let provider = handle.http_provider();
+    let accounts: Vec<_> = handle.dev_wallets().collect();
+    let from = accounts[0].address();
+    let to = accounts[1].address();
+    let amount = handle.genesis_balance().checked_div(2u64.into()).unwrap();
+    // send a dummy transactions
+    let tx = TransactionRequest::new().to(to).value(amount).from(from);
+    let _ = provider.send_transaction(tx, None).await.unwrap().await.unwrap().unwrap();
+
+    let block = provider.get_block_with_txs(1u64).await.unwrap().unwrap();
```

Comment:
Maybe make this `let block: Block<Transaction>`? so that the test also ensures the type